bitwarden_access_manager.py

```diff
@@ -0,0 +1,715 @@
+            access=GroupAccess[data["access"].upper()],
+        )
+
+    def format_toml(self) -> str:
+        return (
+            '{ group_name = "'
+            + self.group_name
+            + '", access = "'
+            + self.access.name.lower()
+            + '" }'
+        )
+
+
+class Collection(NamedTuple):
+    id: str
+    external_id: str
+    group_access: Tuple[GroupCollectionAccess, ...]
+    member_access: Tuple[MemberCollectionAccess, ...]
+
+    def get_id(self) -> str:
+        return self.id
+
+    @staticmethod
+    def from_toml_dict(data: Dict[str, Any]) -> Collection:
+        group_access: Tuple[GroupCollectionAccess, ...] = tuple()
+        if "group_access" in data:
+            group_access = tuple(
+                sorted(
+                    GroupCollectionAccess.from_toml_dict(x)
+                    for x in data["group_access"]
+                )
+            )
+
+        member_access: Tuple[MemberCollectionAccess, ...] = tuple()
+        if "member_access" in data:
+            member_access = tuple(
+                sorted(
+                    MemberCollectionAccess.from_toml_dict(x)
+                    for x in data["member_access"]
+                )
+            )
+
+        return Collection(
+            id=data["collection_id"],
+            external_id=data["external_id"],
+            group_access=group_access,
+            member_access=member_access,
+        )
+
+    def format_toml(self) -> str:
+        result = (
+            "[[collection]]\n"
+            f'collection_id = "{self.id}"\n'
+            f'external_id = "{self.external_id}"\n'
+        )
+
+        member_access_lines = [
+            "  " + a.format_toml() for a in sorted(self.member_access)
+        ]
+        if len(member_access_lines) > 0:
+            result = (
+                result
+                + "member_access = [\n"
+                + ",\n".join(member_access_lines)
+                + ",\n]\n"
+            )
+        else:
+            result = result + "member_access = []\n"
+
+        group_access_lines = [
+            "  " + a.format_toml() for a in sorted(self.group_access)
+        ]
+        if len(group_access_lines) > 0:
+            result = (
+                result
+                + "group_access = [\n"
+                + ",\n".join(group_access_lines)
+                + ",\n]"
+            )
+        else:
+            result = result + "group_access = []"
+
+        return result
+
+
+class BitwardenClient(NamedTuple):
+    connection: HTTPSConnection
+    bearer_token: str
+
+    @staticmethod
+    def new(client_id: str, client_secret: str) -> BitwardenClient:
+        response = requests.post(
+            "https://identity.bitwarden.com/connect/token",
+            headers={"Content-Type": "application/x-www-form-urlencoded"},
+            data={
+                "grant_type": "client_credentials",
+                "scope": "api.organization",
+                "Accept": "application/json",
```

Comment:
Does the `Accept` really go here? This is an http header, shouldn’t it go in `headers`?